src/server.rs

```diff
@@ -87,66 +83,62 @@ impl<'a> SidekiqServer<'a> {
         }
     }
 
-    pub fn attach_handler<T: JobHandler + 'a>(&mut self, name: &str, handle: T) {
-        self.job_handlers.insert(name.into(), Box::new(handle));
-    }
-
-    pub fn attach_middleware<T: MiddleWare + 'a>(&mut self, factory: T) {
-        self.middlewares.push(Box::new(factory));
+    pub fn attach_handler(&mut self, name: &str, handle: impl JobHandler + 'static) {
+        self.job_handlers.insert(name.into(), Arc::new(Box::new(handle)));
     }
 
-    pub fn start(&mut self) {
+    pub async fn start(&mut self) {
         info!("sidekiq is running...");
         if self.queues.len() == 0 {
             error!("queue is empty, exiting");
             return;
         }
-        let (tsx, rsx) = crossbeam_channel::bounded(self.concurrency + 10);
-        let (tox, rox) = crossbeam_channel::bounded(self.concurrency + 10);
+        let (tsx, rsx) = bounded(self.concurrency + 10);
+        let (tox, rox) = bounded(self.concurrency + 10);
         let signal = self.signal_chan.clone();
 
         // start worker threads
-        self.launch_workers(tsx.clone(), rox.clone());
+        self.launch_workers(tsx.clone(), rox.clone()).await;
 
         // controller loop
-        let (tox2, rsx2) = (tox.clone(), rsx.clone()); // rename channels b/c `select!` will rename them below
-        let clock = tick(Duration::from_secs(2)); // report to sidekiq every 5 secs
+        let (tox2, rsx2) = (tox.clone(), rsx.clone());
+        let mut clock = time::interval(Duration::from_secs(2));
         loop {
-            if let Err(e) = self.report_alive() {
+            if let Err(e) = self.report_alive().await {
                 error!("report alive failed: '{}'", e);
             }
             select! {
-                recv(signal) -> signal => {
+                biased;
```

Comment:
Both the server and worker `select!` loops set `biased` to preserve the original behavior where each entry is attempted in order. https://docs.rs/tokio/latest/tokio/macro.select.html#fairness